aiidalab_widgets_base/viewers.py

```diff
@@ -168,44 +174,48 @@ class _StructureDataBaseViewer(ipw.VBox):
     DEFAULT_SELECTION_RADIUS = 6
     DEFAULT_SELECTION_COLOR = "green"
 
-    def __init__(self, configure_view=True, **kwargs):
+    def __init__(
+        self,
+        configure_view=True,
+        configuration_tabs=["Selection", "Appearance", "Cell", "Download"],
+        default_camera="orthographic",
+        **kwargs,
+    ):
         # Defining viewer box.
 
-        # 1. Nglviwer
+        # Nglviwer
         self._viewer = nglview.NGLWidget()
-        self._viewer.camera = "orthographic"
+        self._viewer.camera = default_camera
         self._viewer.observe(self._on_atom_click, names="picked")
         self._viewer.stage.set_parameters(mouse_preset="pymol")
 
-        # 2. Camera type.
-        camera_type = ipw.ToggleButtons(
-            options={"Orthographic": "orthographic", "Perspective": "perspective"},
-            description="Camera type:",
-            value="orthographic",
-            layout={"align_self": "flex-end"},
-            style={"button_width": "115.5px"},
-            orientation="vertical",
-        )
-
-        def change_camera(change):
+        view_box = ipw.VBox([self._viewer])
 
-            self._viewer.camera = change["new"]
+        configuration_tabs_map = {
+            "Selection": self._selection_tab(),
+            "Appearance": self._appearance_tab(),
+            "Cell": self._cell_tab(),
+            "Download": self._download_tab(),
+        }
 
-        camera_type.observe(change_camera, names="value")
-        view_box = ipw.VBox([self._viewer, camera_type])
+        if configure_view is not True:
+            warnings.warn(
+                "`configure_view` is deprecated, please use `configuration_tabs` instead.",
+                DeprecationWarning,
+            )
+            if configure_view is False:
+                configuration_tabs.clear()
 
         # Constructing configuration box
-        if configure_view:
+        if len(configuration_tabs) != 0:
             configuration_box = ipw.Tab(
                 layout=ipw.Layout(flex="1 1 auto", width="auto")
             )
             configuration_box.children = [
```

Comment:
What should happen if the user provided a non-existing tab name? Should it fail as it does now, or should there be a warning message?

I am fine with any approach. Just mentioning it here, as it is worth considering.

Reply:
@yakutovicha good question. In general I am often unsure about how to handle / propagate errors. In this case I'd say this would be a programming error and it is okay to crash.

Reply:
ok, fine with me.